### PR TITLE
fix(lemon-ui): LemonInputSelect 2.1

### LINKED ISSUE #23832: Bug Report: Feature flags - when editing matching criteria, clicking a value deletes it
## Bug description

When creating or editing a feature flag, in the form field for entering values, when you hover over a value you've already entered, you see a tooltip saying `Click to edit`, but clicking on the value deletes it.  

The expectation is that clicking the value will allow you to edit it, and that deleting the value would require clicking on the `X` on the rightside of the value:

CleanShot 2024-07-18 at 10 12 53

## How to reproduce

1. Create or edit a feature flag
2. Keep the default for 'Match by' `Users`
3. Select a person property (e.g. `Distinct ID`)
4. Enter/select one or more values
5. Hover over a value and note the `Click to edit` tooltip
6. Click the value, notice it is deleted instead of being made editable.

## Additional context
From: https://posthoghelp.zendesk.com/agent/tickets/15845

### Debug info

```shell
Session: https://us.posthog.com/project/sTMFPsFhdP1Ssg/replay/0190c6c7-f775-72c7-8464-b3b6c10db818?t=970
Admin: http://go/adminOrgUS/018e0d19-ade7-0000-d4af-dbad1dcf0ab8 (project ID 56976)
Sentry: http://go/sentryUS/56976
```

### LINKED ISSUE #24317: Bug Report: Insights - when values for a field in a filter, clicking a value deletes it, with or without the `alt / option` modifier
## Bug description

When creating or editing a filter on an insight, in the form field for entering values when you hover over a value you've already entered, you see a tooltip saying `⌥ + Click to edit`, but `⌥ + click` on the value deletes it (clicking on it without a modifier key also deletes it) 

(see the video at the link in [this ticket](https://posthoghelp.zendesk.com/agent/tickets/16739))

The expectation is that `⌥ + click` on the value will allow you to edit the value, and that you should have to click on the `X` to delete the value.

## How to reproduce

1. Create or edit a trends insight
2. Keep the default for 'Match by' Users
3. Add an `A` series (e.g. a pageview)
4. Add a filter to the series (e.g. for a `CSS Selector` `= equals` )
5. Enter/select one or more values for the filter
6. Hover over a value and note the tooltip showing `⌥ + Click to edit`
7. Hold `⌥` while clicking a value you entered, notice the is deleted instead of being made editable. 

## Additional context

from: https://posthoghelp.zendesk.com/agent/tickets/16739

note: related to, but different from https://github.com/PostHog/posthog/issues/23832

### Debug info

```shell
Report event: http://go/ticketByUUID/8db8fd24-8019-4263-bf82-8e8553253e6f
Session: https://us.posthog.com/project/sTMFPsFhdP1Ssg/replay/01913858-aa36-7646-a2a9-1774ae92c724?t=2211
Location: https://us.posthog.com/project/82866/insights/fIEeBpVm/edit
Admin: http://go/adminUS/ozan@perculus.com (organization ID 019112e6-d493-0000-b2be-98ba8a249407: Perculus, project ID 82866: Perculus)
Sentry: http://go/sentryUS/82866
```

### PR DESCRIPTION
**Pull request by Twixes**
## Changes

#24321 improved `LemonInputSelect`, but it remained buggy around removing and editing the already-selected options, which users rightly point out. This resolves #24317, resolves #23832, makes the input itself autosized, and revamps the editing experience to avoid the value shifting around in a surprising way:

### Before
before

### After
after

### CODE REVIEW

File: frontend/__snapshots__/scenes-app-feature-flags--edit-feature-flag--dark.png
Comment (Twixes) [inline frontend/__snapshots__/scenes-app-feature-flags--edit-feature-flag--dark.png:1]: A lot of these changes, as LemonSnack is now very slightly less tall to better align with input and button heights

**Comment by Twixes** [comment]
Actually also really liked the idea from https://github.com/PostHog/posthog/issues/24317#issuecomment-2296435385 to make clicking on the _text_ initiate editing. This in fact simplifies things. Here's the final iteration up for review:

**Review by skoob13 — APPROVED** [review]
I like that you've removed the hotkey. It's much easier now. I left a few comments about React hooks.

One small problem is that the input blurs when there isn't a filled text and you press the backspace key. I'm not sure if it is an intended behavior.

File: frontend/src/lib/lemon-ui/LemonInput/RawInputAutosize.tsx [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        node.style.fontStyle = styles.fontStyle
+        node.style.letterSpacing = styles.letterSpacing
+        node.style.textTransform = styles.textTransform
+    }
+
+    useLayoutEffect(() => {
+        if (inputRef.current) {
+            setInputStyles(getComputedStyle(inputRef.current))
+        }
+    }, [inputRef.current])
```
[the remark this answers is not in the document]
Comment (Twixes) [reply frontend/src/lib/lemon-ui/LemonInput/RawInputAutosize.tsx:33]: I do not know, as this was all written by @daibhin and I just moved it from `frontend/src/lib/components/EditableField/EditableField.tsx` to its own file 😄

File: frontend/src/lib/lemon-ui/LemonInput/RawInputAutosize.tsx [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        let newInputWidth
+        if (inputProps.placeholder && !inputProps.value) {
+            newInputWidth = Math.max(sizerRef.current.scrollWidth, placeHolderSizerRef.current.scrollWidth) + 2
+        } else {
+            newInputWidth = sizerRef.current.scrollWidth + 2
+        }
+        if (newInputWidth !== inputWidth) {
+            setInputWidth(newInputWidth)
+        }
+    }, [sizerRef.current, placeHolderSizerRef.current, inputProps.placeholder, inputProps.value])
```
[the remark this answers is not in the document]
Comment (Twixes) [reply frontend/src/lib/lemon-ui/LemonInput/RawInputAutosize.tsx]: Added `inputWidth`. Although the refs I'm not sure about. I haven't written this, but I have seen cases where `fooRef.current` made for a key dependency (if a bit smelly). Will leave in

**Comment by Twixes** [comment]
That's odd @skoob13, I can't reproduce the Backspace behavior…

**Comment by anirudhpillai** [comment]
This works so much better now!
Just a small thing with highlighting what's selected, when you type a substring what's already selected, it highlights multiple rows.
image
image

**Comment by Twixes** [comment]
@anirudhpillai Good point, that's something I was unsure about, as I carried over this way of highlighting selected items from the original LemonInputSelect – but now it's redundant. Will rely on the checkbox only, so that keyboard navigation is obvious.

**Comment by skoob13** [comment]
It's probably unrelated, but it's a very annoying behavior. When I paste a long string like:
```
Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/121.0.6167.57 Safari/537.36
```
The input splits it into two options.

Repro:

**Comment by Twixes** [comment]
Yeah, it's been that way since the beginning @skoob13. That's handy if you have multiple values you'd like to add at once, but annoying in cases like the one you recorded. Will leave that as is in this PR though

**Pull request merged**